File: sdk/.net/ardvro.sdk.net/component/kenviro/controller/ProbabilisticController.cs
namespace ardvro.sdk.net.component.kenviro.controller;

using ardvro.core.ext.connection;
using ardvro.core.lib.common;

public class ProbabilisticController
{
    private IConnection _connection;
    public ProbabilisticController(IConnection connection)
    {
        _connection = connection;
    }

    public void Compute (String probabilisticType, double[][]inputs, double[] weights, Action<double[][]> callback)
    {
        var args = new Flexible();
        args.Add("ProbabilisticType", probabilisticType);
        args.Add("Inputs", inputs);
        args.Add("Weights", weights);
        _connection.Submit(args, "ardvro/component/kenviro/Probabilistic/Compute", callback);
    }

}

File: sdk/web/component/kenviro/controller/ProbabilisticController.js
var ProbabilisticController = function ProbabilisticController(cfg)
{
    let ctrl = {};

    ctrl.Compute = function (probabilisticType, inputs, weights, callback)
    {
        let args = {
            ProbabilisticType: probabilisticType,
            Inputs: inputs,
            Weights: weights
        };
        cfg.Connector.Submit(args, "ardvro/component/kenviro/Probabilistic/Compute", function (box)
        {
            callback != null ? callback(box == null ? null : box.Data) : null;
        });
    };


    return ctrl;
}